### PR TITLE
Fix top nav children not rendered

### LINKED ISSUE #3725: No navigation nodes rendered, if children are empty of one node
**Description**
If the children of a navigation node are empty, no further nodes are rendered.
Reason behind this is, that a user might visit the page but has no access to any of these nodes, therefore they are not delivered by the backend. But there should be a button in the top navigation that brings him back to an overall preselection so he can select an other customer where he might have permissions. 

**Expected result**
Other nodes to be rendered anyways

**Actual result**
Not rendered

**Steps to reproduce**

Use this config e.g. in the playground. The "Some Action" is not displayed.

```
Luigi.setConfig({
    navigation: { 
            validWebcomponentUrls:['.*?'],
            nodes: [{ 
                pathSegment: 'home', 
                label: 'h', 
                hideFromNav: true, 
                children: [] 
            },{ 
                pathSegment: 'foo', 
                label: 'Some Action',
                icon: 'favorite-list',
                viewUrl: '/examples/microfrontends/multipurpose.html',
                hideSideNav: true,
                context: {
                    title: 'Left navigation hidden',
                    content: 'for pages needing more space or wanting to handle navigation internally'  
                }
            },{ 
                pathSegment: 'help', 
                label: 'Help',
                icon: 'sys-help',
                viewUrl: '/examples/microfrontends/multipurpose.html',
                context: {
                    title: 'Help Section',
                    content: 'Find some useful links on the left'  
                },
                children: [{
                    label: 'Back',
                    link: '/',
                    icon: 'nav-back'
                },{
                    label: 'Luigi Github Page',
                    externalLink : { 
                        url: 'https://github.com/SAP/luigi'
                    }
                },{
                    label: 'Fundamental Library',
                    externalLink : { 
                        url: 'https://sap.github.io/fundamental-styles'
                    }
                },{
                    label: 'Fundamental Icons',
                    externalLink : { 
                        url: 'https://sap.github.io/fundamental-styles/components/icon.html'
                    }
                }]
            }],
            productSwitcher: {
                items: [{
                    icon: 'https://raw.githubusercontent.com/SAP/luigi/main/website/landingpage/public/assets/img/logos/sap.svg',
                    label: 'SAP homepage',
                    externalLink: {
                      url: 'https://www.sap.com',
                      sameWindow: false
                    }
                },{
                    icon: 'https://svelte.technology/favicon.ico',
                    label: 'Svelte',
                    externalLink: {
                      url: 'https://svelte.technology',
                      sameWindow: false
                    }
                }]
            } 
        }, 
        
        // uncomment if you want to use our mock oidc idp
        /**
        auth: {
            // We have registered the following provider at the window object:
            // OAuth2 Implicit Grant: window.LuigiAuthOAuth2 - Docs: https://docs.luigi-project.io/docs/authorization-configuration?section=oauth2-implicit-grant-configuration
            // OIDC Implicit Grant: window.LuigiAuthOIDC - Docs: https://docs.luigi-project.io/docs/authorization-configuration/?section=openid-connect-configuration
 
            use: 'myOAuth2',
            myOAuth2: {
                idpProvider: window.LuigiAuthOAuth2,
                authorizeUrl: '/auth/idpmock/implicit.html',
                logoutUrl: '/auth/idpmock/logout.html',
                post_logout_redirect_uri: '/auth/logout.html',
                authorizeMethod: 'GET',
                oAuthData: {
                  client_id: 'egDuozijY5SVr0NSIowUP1dT6RVqHnlp',
                  redirect_uri: '/auth/callback.html'
                }
            }
        },
        */
        routing: { 
            useHashRouting: true 
        }, 
        lifecycleHooks: {
            luigiAfterInit: () => {
                console.log('Luigi initialized.');
            }
        },
        settings: { 
            responsiveNavigation: 'semiCollapsible',
            header: { 
                logo: 'img/luigi.svg', 
                title: 'Luigi Fiddle'
            },
            burgerTooltip: {
                navExpanded: 'Collapse navigation',
                navCollapsed: 'Expand navigation'
            }
        }
    });    
```

**Troubleshooting**

None

### PR DESCRIPTION
**Pull request by JohannesDoberer**
Fixes #3725

### CODE REVIEW

**Review by ndricimrr — APPROVED** [review]
Tested and works as expected. I would create a follow up to have a test for this scenario written.

**Comment by JohannesDoberer** [comment]
@ndricimrr I'll add a test scenario to this pr.

File: test/e2e-test-application/cypress/e2e/tests/0-js-test-app/js-test-app-navigation.cy.js [hunk 1]
```diff
@@ -653,4 +653,32 @@ describe('JS-TEST-APP', () => {
       });
     });
   });
+
+  describe.only('First topNav node has no viewURL and empty children', () => {
```
Comment (ndricimrr) [inline test/e2e-test-application/cypress/e2e/tests/0-js-test-app/js-test-app-navigation.cy.js:657]:
```suggestion
  describe('First topNav node has no viewURL and empty children', () => {
```

Comment (JohannesDoberer) [reply test/e2e-test-application/cypress/e2e/tests/0-js-test-app/js-test-app-navigation.cy.js:657]: Thx 👍 🤦

File: client/src/linkManager.js [hunk 2]
```diff
@@ -278,7 +278,7 @@ export class linkManager extends LuigiClientBase {
    */
   fromClosestContext() {
     const hasParentNavigationContext =
-      this.currentContext && this.currentContext.context.parentNavigationContexts.length > 0;
+      this.currentContext && this.currentContext.context?.parentNavigationContexts?.length > 0;
```
Comment (walmazacn) [inline client/src/linkManager.js:281]: This line might be even shorter:
`this.currentContext?.context?.parentNavigationContexts?.length`

**Review by walmazacn — APPROVED** [review]
Everything looks good

**Review by ndricimrr — APPROVED** [review]
Looks good 👍

**Review by ndricimrr — APPROVED** [review]
tested again and works

**Pull request merged**